aspects/src/main/java/org/springframework/security/authorization/method/aspectj/ExceptionWrappingMethodInterceptor.aj

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method.aspectj;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+
+final class ExceptionWrappingMethodInterceptor {
```

Comment:
Alternatively (I think preferably), there could be an `AbstractMethodInterceptorAspect` which performs Exception handling and shared logic for all the Aspects. Then, the subclasses could provide the pointcuts.

Reply:
I think another alternative is to rename to `ExceptionUtils#handleChecked` that takes a risky lambda and adapts any checked exception into an unchecked one. Then the code can instead do:

```java
return ExceptionUtils.handleChecked(() -> this.securityInterceptor.invoke(invocation));
```

OOC, Spring Security seems to often prefer delegation over inheritance. What makes inheritance seem more preferable in this case? When I tried it out, the one thing that bugged me is that the subclass's methods -- because they were the ones with the pointcut annotations -- needed to call into the superclass, which historically has been a code smell for me.

Reply:
nm, your link in a later comment regarding `AbstractCacheAspect` addresses my concern.